src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -99,12 +89,12 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public string Name { get; }
         public bool IsValueType { get; }
         public bool IsEnum { get; }
-        public bool IsLiteral { get; init; }
-        public Constant? Literal { get; init; }
-        public bool IsUnion { get; }
-        public IReadOnlyList<CSharpType> UnionItemTypes { get; } = Array.Empty<CSharpType>();
+        public bool IsLiteral => Literal is not null;
+        public Constant? Literal { get; private init; }
+        public bool IsUnion => UnionItemTypes.Count > 0;
+        public IReadOnlyList<CSharpType> UnionItemTypes { get; private init; } = Array.Empty<CSharpType>();
```

Comment:
`IsLiteral` and `Literal` actually stand for the same thing, they should not be able to set separately because it is very likely to misuse.